gridsuite/dynamic-simulation-server
# Dynamic Simulation - Unit test with test container

Use image from https://github.com/powsybl/java-dynawo-docker for test container with command

**See test container with command** : https://www.testcontainers.org/features/commands/

**In  this PR:**

- Change the pom.xml, using <jib.from.image>powsybl/java-dynawo:1.4.0</jib.from.image> instead of <jib.from.image>gridsuite/java-dynawo</jib.from.image>
- In test/resources/powsybl/config/test/config.yml, set dynawaltz.homeDir = /dynawo which is the directory of dynawo inside the image powsybl/java-dynawo
- Do not download directly dynawo distribution in build workflow, DockerLocalComputationManager will download the java-dynawo image
- Adapt unit test in order to test only the solver SIM.. because the solver IDA at moment (dyanwo 1.4.0) is nondeterministic over different OS, i.e. Debian (java-dynawo which bases on powsybl/java:1.0.0 (Debian)) vs Ubuntu 

**Steps to test new source code with docker-compose:**

- Launch all services by docker-compose from deployment/docker-compose/study: **docker-compose up**
- Stop the current running image from deployment/docker-compose/study : **docker-compose stop dynamic-simulation-server** 
- Go inside the root directory of checkout project: for example /home/phamquy/Projects/gridsuite/dynamic-simulation-server
- Launch following command to build docker image: **mvn clean -Dpowsybl.docker.install install -DskipTests**, the result image is built with  powsybl/java-dynawo:1.4.0 by jib 
- Check whether the new latest image is generated :  **docker image ls | grep dynamic-simulation-server**
- Relaunch the latest image from deployment/docker-compose/study : **docker-compose up dynamic-simulation-server**
- Run gridstudy-app to test the dynamic simulation from the checkout project: **npm start**

**Steps to test new source code in IDEA:**

- Download Dynawo distribution: https://github.com/dynawo/dynawo/releases/download/v1.4.0/Dynawo_Linux_v1.4.0.zip
- Unzip in a directory : /home/yourusername/dynawo for example
- Configure **/home/yourusername/.itool/config.yml**, set
`dynamic-simulation:
    default-impl-name: "DynaWaltz"
    
dynawaltz:
    homeDir: /home/yourusername/dynawo
    debug: true
`
![image](https://user-images.githubusercontent.com/117309322/234979543-70842443-b20b-449b-8394-558c20a8889f.png)

- Launch DynamicSimulationApplication for the server

**Steps to run unit tests:**

- Launch unit tests inside IDEA ... or by command : **mvn clean test**, nothing to configure thanks to test container
